payloadcms/payload
# fix(next): only set payload theme preference headers within '/admin' namespace

When these headers are set globally, they interfere with the lighthouse CI tool. It forces a redirect that adds 600-800ms to the total load time on mobile.

By only setting the headers for the `/admin` section of the site, which is where they are used, it resolves the problem for other sections of the site where performance is critical.

## Description



- [x] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change



## Steps to test

1. Pull down the latest version of the [payload 3.0 demo](https://github.com/payloadcms/payload-3.0-demo)
2. Run the site in prod mode: `pnpm build && pnpm start`
3. Run a lighthouse audit using the cli tool: `npx lighthouse@latest http://localhost:3000 --view --chrome-flags="--headless" --verbose`
4. Observe this line in the report:
    <img width="793" alt="Screenshot 2024-09-05 at 2 33 11 PM" src="https://github.com/user-attachments/assets/be28e1ae-0647-447f-8da5-32094aaa7560">

## Steps to confirm the fix

1. Open the file: `node_modules/@payloadcms/next/dist/withPayload.js`
2. Apply the change from this PR by changing `source: '/:path*'` to `source: '/admin/:path*'`
3. Re-run build and run the site: `pnpm build && pnpm start`
4. Re-run the lighthouse check: `npx lighthouse@latest http://localhost:3000 --view --chrome-flags="--headless" --verbose`
5. Observe that the redirect perfomance penalty has been resolved
